ssg/build_remediations.py

```diff
@@ -332,8 +332,7 @@ def _get_rule_reference(self, ref_class):
     def update_when_from_rule(self, to_update):
         additional_when = ""
         if self.associated_rule.platform == "machine":
-            additional_when = ('ansible_virtualization_role != "guest" '
```

Comment:
@matejak Was the goal of `ansible_virtualization_role != "guest"` to prevent it from running on VMs?
I remember it was related to podman containers, but my memory is not clear.

Maybe at the time it was not clear what would be set in a podman container.
Found this [ansible issue](https://github.com/ansible/ansible/pull/64981/files), and seems that it will be fixed soon.

Reply:
I think that Ansible was broken in the same way as it is now - it doesn't even list the system as guest - Ansible inside a podman container thinks that it is a `kvm` host.
When Ansible detects that it runs in docker, it sets role to `guest` *and* type to `docker`.
The `when` clause determines when to run the remediation. Prior to this PR, anything else than `docker` and `guest` **at the same time** resulted in that condition to be true, therefore executing the task.